core/src/main/java/com/google/instrumentation/trace/TraceService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+/**
+ * Class that holds the implementation instances for {@link Tracer} and {@link
+ * BinaryPropagationHandler}.
+ *
+ * <p>This implementation of this class is loaded using reflection by the {@link Trace}.
```

Comment:
is this intentionally the only pattern? or is it likely that in dependency injection, like Guice or Dagger, one would return a TraceService as an object graph that includes the tracer and propagation handler? Or would we expect in DI, that they just configure and inject Tracer and propagation handler directly (which implies TraceService is only used for JVM/classloader singletons)